internal/packages/changelog/yaml.go

```diff
@@ -0,0 +1,160 @@
+// you may not use this file except in compliance with the Elastic License.
+
+package changelog
+
+import (
+	"bytes"
+
+	"github.com/Masterminds/semver"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// PatchYAML looks for the proper place to add the new revision in the changelog,
+// trying to conserve original format and comments.
+func PatchYAML(d []byte, patch Revision) ([]byte, error) {
+	var nodes []yaml.Node
+	err := yaml.Unmarshal(d, &nodes)
+	if err != nil {
+		return nil, err
+	}
+
+	patchVersion, err := semver.NewVersion(patch.Version)
+	if err != nil {
+		return nil, err
+	}
+
+	patched := false
+	var result []yaml.Node
+	for _, node := range nodes {
+		if patched {
+			result = append(result, node)
+			continue
+		}
+
+		var entry Revision
+		err := node.Decode(&entry)
+		if err != nil {
+			result = append(result, node)
+			continue
+		}
+
+		foundVersion, err := semver.NewVersion(entry.Version)
+		if err != nil {
+			return nil, err
+		}
+
+		if foundVersion.GreaterThan(patchVersion) {
+			return nil, errors.New("cannot add change to old version")
+		}
+
+		var newNode yaml.Node
+		if patchVersion.Equal(foundVersion) {
+			// Add the change to current entry.
+			entry.Changes = append(patch.Changes, entry.Changes...)
+			err := newNode.Encode(entry)
+			if err != nil {
+				return nil, err
+			}
+			// Keep comments of the original node.
+			newNode.HeadComment = node.HeadComment
+			newNode.LineComment = node.LineComment
+			newNode.FootComment = node.FootComment
+			// Quote version to keep common style.
+			setYamlMapValueStyle(&newNode, "version", yaml.DoubleQuotedStyle)
+			result = append(result, newNode)
+			patched = true
+			continue
+		}
+
+		// Add the change before first entry
+		err = newNode.Encode(patch)
+		if err != nil {
+			return nil, err
+		}
+		// If there is a comment on top, leave it there.
+		if node.HeadComment != "" {
+			newNode.HeadComment = node.HeadComment
+			node.HeadComment = ""
+		}
+		// Quote version to keep common style.
+		setYamlMapValueStyle(&newNode, "version", yaml.DoubleQuotedStyle)
+		result = append(result, newNode, node)
+		patched = true
+	}
+
+	if !patched {
+		return nil, errors.New("changelog entry was not added, this is probably a bug")
+	}
+
+	d, err = yaml.Marshal(result)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to encode resulting changelog")
+	}
+
+	return d, nil
+}
+
+func SetManifestVersion(d []byte, version string) ([]byte, error) {
+	var node yaml.Node
+	err := yaml.Unmarshal(d, &node)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to decode manifest")
+	}
+
+	// Manifest is a document, with a single element, that should be a map.
+	if len(node.Content) == 0 || node.Content[0].Kind != yaml.MappingNode {
+		return nil, errors.Wrap(err, "unexpected manifest content")
+	}
+
+	setYamlMapValue(node.Content[0], "version", version)
+
+	var buf bytes.Buffer
+	encoder := yaml.NewEncoder(&buf)
+	encoder.SetIndent(2)
```

Comment:
FYI we have the [formatter](https://github.com/elastic/elastic-package/blob/main/internal/formatter/yaml_formatter.go) in the source code, so it would be better not to introduce another `SetIndent`  call :)